src/main/java/Ultron/Commands/TaskAllocator.java

```diff
@@ -0,0 +1,87 @@
+package ultron.commands;
+
+import ultron.Storage;
+import ultron.TaskList;
+import ultron.exceptions.ExceptionType;
+import ultron.exceptions.UltronException;
+import ultron.tasks.Task;
+import ultron.ui.UI;
+
+public final class TaskAllocator extends Command {
+
+    /**
+     * Store TaskCommand corresponding to the command.
+     */
+    private final TaskCommand taskCommand;
+
+    /**
+     * Allocates the correct task.
+     *
+     * @param command   Command given.
+     * @param arguments Arguments given.
+     * @throws UltronException if the command is invalid.
+     */
+    public TaskAllocator(final String command,
+                         final String arguments) throws UltronException {
+
+        //Call the superclass
+        super(false, arguments);
+
+        try {
+            //Get the state
+            this.taskCommand = TaskCommand.valueOf(command.toLowerCase());
+
+        } catch (IllegalArgumentException e) {
+
+            //Throw a Duke exception
+            throw new UltronException(command,
+                ExceptionType.INVALID_COMMAND);
```

Comment:
Indentation for wrapped lines should be 8 spaces.

Reply:
I see. Thank you!